src/main/kotlin/no/nav/k9/inngaende/oppslag/BarnOppslag.kt

```diff
@@ -1,41 +1,85 @@
 package no.nav.k9.inngaende.oppslag
 
-import no.nav.k9.utgaende.gateway.AktoerRegisterV1Gateway
-import no.nav.k9.utgaende.gateway.TpsProxyV1Gateway
-import no.nav.k9.utgaende.rest.AktørId
-import no.nav.k9.utgaende.rest.TpsBarn
+import io.ktor.util.*
+import no.nav.k9.clients.pdl.generated.HentBarn
+import no.nav.k9.clients.pdl.generated.HentBarn.AdressebeskyttelseGradering.UGRADERT
+import no.nav.k9.utgaende.gateway.PDLProxyGateway
+import java.lang.IllegalStateException
+import java.time.LocalDate
 
 internal class BarnOppslag(
-    private val aktoerRegisterV1Gateway: AktoerRegisterV1Gateway,
-    private val tpsProxyV1Gateway: TpsProxyV1Gateway
+    private val pdlProxyV1Gateway: PDLProxyGateway,
 ) {
 
+    @KtorExperimentalAPI
     internal suspend fun barn(
-        ident: Ident,
-        attributter: Set<Attributt>
-    ) : Set<Barn>? {
+        barnasIdenter: List<Ident>,
+        attributter: Set<Attributt>,
+    ): Set<Barn>? {
         if (!attributter.etterspurtBarn()) return null
 
-        val tpsBarn = tpsProxyV1Gateway.barn(
-            ident = ident,
-            attributter = attributter
-        ) ?: return null
-
-        return tpsBarn
-            .filter { it.dødsdato == null }
-            .map {
-                Barn(
-                    tpsBarn = it,
-                    aktørId = aktoerRegisterV1Gateway.aktørId(
-                        ident = it.ident,
-                        attributter = attributter
+        return when {
+            barnasIdenter.isEmpty() -> null
+            else -> pdlProxyV1Gateway.barn(barnasIdenter)
+                .filter { it.person != null }
+                .filter { it.person!!.ikkeErBeskyttet() }
+                .filter { it.person!!.erILive() }
+                .map {
+                    Barn(
+                        pdlBarn = it.tilPdlBarn(),
+                        aktørId = pdlProxyV1Gateway.aktørId(
+                            ident = Ident(it.ident),
+                            attributter = attributter
+                        )?.tilAktørId()
                     )
-                )
-            }.toSet()
+                }.toSet()
+        }
     }
 }
 
+/**
+ * Adressebeskyttelse er ofte tomt når det er ugradert.
+ *
+ * https://navikt.github.io/pdl/#_adressebeskyttelse
+ */
+private fun HentBarn.Person.ikkeErBeskyttet(): Boolean = adressebeskyttelse.isEmpty() ||
```

Comment:
Burde den heller hete "ikkeBeskyttet" eller "erIkkeBeskyttet"?

Reply:
Spørs hvordan du leser denne tenker jeg.
Grunnen til at det står ikkeErBeskyttet framfor erIkkeBeskyttet er pga. kodestil.

Jeg syns at `.filter { it.person!!.ikkeErBeskyttet() }` er forståelig da :P